go/tasks/plugins/hive/execution_state.go

```diff
@@ -84,6 +90,9 @@ func HandleExecutionState(ctx context.Context, tCtx core.TaskExecutionContext, c
 	case PhaseSubmitted:
 		newState, transformError = MonitorQuery(ctx, tCtx, currentState, executionsCache)
 
+	case PhaseWriteOutputFile:
```

Comment:
why is this a separate phase?

Reply:
The bit of code that monitors the query status currently is done inside the cache handler, that just runs in goroutines in the background.  For something like writing to S3, it made more sense to make it an explicit plugin state.